trieste/models/model_interfaces.py

```diff
@@ -26,6 +28,22 @@
 class ProbabilisticModel(ABC):
     """ A probabilistic model. """
 
+    @abstractmethod
+    def optimize(self, dataset: Dataset) -> None:
+        """
+        Optimize the model.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def update(self, dataset: Dataset) -> None:
+        """
+        Update the model given the specified ``dataset``. Does not train the model.
+
+        :param dataset: The data with which to update the model.
+        """
+        raise NotImplementedError
+
```

Comment:
I don't know if this you meant to do this, but this undoes the previous model PR